diff --git a/node_modules/vscode-apollo-relay/dist/generateConfig.js b/node_modules/vscode-apollo-relay/dist/generateConfig.js
index 7ada1b9..7521ea5 100644
--- a/node_modules/vscode-apollo-relay/dist/generateConfig.js
+++ b/node_modules/vscode-apollo-relay/dist/generateConfig.js
@@ -8,7 +8,7 @@ var __spreadArrays = (this && this.__spreadArrays) || function () {
 };
 exports.__esModule = true;
 var path = require("path");
-var RelayCompilerMain_1 = require("relay-compiler/lib/RelayCompilerMain");
+// var RelayCompilerMain_1 = require("relay-compiler/lib/RelayCompilerMain");
 var dependencies_1 = require("./dependencies");
 var generateDirectivesFile_1 = require("./generateDirectivesFile");
 var RelayKnownArgumentNames_1 = require("./RelayKnownArgumentNames");
@@ -36,7 +36,9 @@ function loadRelayConfig() {
 }
 function generateConfig() {
     var relayConfig = loadRelayConfig();
-    var languagePlugin = RelayCompilerMain_1.getLanguagePlugin((relayConfig && relayConfig.language) || "javascript");
+    // The old version of relay Reaction is still on does not export this function.
+    // var languagePlugin = RelayCompilerMain_1.getLanguagePlugin((relayConfig && relayConfig.language) || "javascript");
+    var languagePlugin = { inputExtensions: ["js", "jsx"]};
     var directivesFile = generateDirectivesFile_1.generateDirectivesFile();
     var includesGlobPattern = function (inputExtensions) { return "**/*.{graphql," + inputExtensions.join(",") + "}"; };
     var config = {
